client/client_test.go

```diff
@@ -215,30 +213,30 @@ func TestCaLoading(t *testing.T) {
 	assert.True(t, oursOK)
 }
 
+type nullSystemCert struct{}
+
+func (nullSystemCert) GetSystemCertPool() (*x509.CertPool, error) {
+	return x509.NewCertPool(), nil
+}
+
 func TestEmptySystemCertPool(t *testing.T) {
 	version := runtime.Version()
 	if strings.HasPrefix(version, "1.6") || strings.HasPrefix(version, "1.7") || strings.HasPrefix(version, "1.8") {
 		// Environment variable not included until version 1.9. Therefore skipping this test.
 		t.SkipNow()
 	}
 
-	tmpdir, err := ioutil.TempDir("", "nocertsfolder")
-	assert.NoError(t, err)
-
-	// Fake the environment variables, to override ssl-cert lookup
-	err = os.Setenv("SSL_CERT_DIR", tmpdir)
-	assert.NoError(t, err)
-
-	err = os.Setenv("SSL_CERT_FILE", tmpdir+"idonotexist.crt") // fakes a non existing cert-file
-	assert.NoError(t, err)
-
 	conf := Config{
 		ServerCert: "server.crt",
 	}
 
-	certs, err := loadServerTrust(conf)
-	assert.NoError(t, err)
-	assert.NotZero(t, certs)
```

Comment:
If I understood your code and commit message, with the previous implementation this test should have been expecting `Error`?

Reply:
Well, kind of. It's clear that no one ever tested the failure case. `SSL_CERT_DIR` actually makes no difference in this test, because `ServerCert: "server.crt` is a valid certificate, and hence you will get `NoError` no matter what you put there. I guess it was attempting to verify that this would always be successful.

But it did not test what would happen if both certificate sources are empty, which is the new case I added below. This would have been `Error` with the old code, but is now `len == 0` instead.

And yet a complicating factor is that using `NotZero` was useless here, as even a failed call to `SystemCertPool` returns a non-nil (but empty) map, and hence `NotZero` would return true in all cases.